### PR TITLE
TBD-5491: fix

### PR DESCRIPTION
**Pull request by mtuhai**
**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [X] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [X] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [X] Docs have been added / updated (for bug fixes / features) ?
- [X] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
Java mapping is being overrided in tSqoopImport if 
https://jira.talendforge.org/browse/TBD-5491

**What is the new behavior?**
Java mapping is preserved

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet [hunk 1]
```diff
@@ -218,7 +225,18 @@
 		} 
 	} 
 %>
+
 	sqoopOptions_<%=cid%>.loadProperties(additionalProperties_<%=cid%>);
+	
+	//restoring java/hive mapping because SqoopOptions.loadProperties() rewrites it completely
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: question about this PR, 
before 
1. we can set MapColumnJava/Hive in JAVA_TYPE_MAPPING/HIVE_TYPE_MAPPING table
2. we can set MapColumnJava/Hive in ADDITIONAL_JAVA table
if we set in both, the setting in 2 will override the setting in 1.

but with this modification, the above behavior break

File: main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet [hunk 2]
```diff
@@ -206,19 +209,44 @@
 				break; 
 			}
             		String currentArgument = additionalMap.get("ADDITIONAL_ARGUMENT");
+            		String currentValue = additionalMap.get("ADDITIONAL_VALUE");
+            		
+            		if (currentArgument.startsWith("\"map.column.java.")) {
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: consider `context` value for ADDITIONAL_ARGUMENT

Comment (mtuhai) [reply main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: What do you mean? 'additionalMap' already contains parsed context values (see ln 44)

Comment (bchen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: this line is javajet code, impossible to get runtime value, could you try context value in the table to see if it works?

File: main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet [hunk 3]
```diff
@@ -218,7 +218,34 @@
 		} 
 	} 
 %>
+
+	java.util.Properties storedJavaMapping_<%=cid%> = sqoopOptions_<%=cid%>.getMapColumnJava();
+	java.util.Properties storedHiveMapping_<%=cid%> = sqoopOptions_<%=cid%>.getMapColumnHive();
+
+	for (java.util.Map.Entry entry: additionalProperties_<%=cid%>.entrySet()) {
+		String curKey = (String)entry.getKey();
+		String curValue = (String)entry.getValue();
```
[the remark this answers is not in the document]
Comment (mtuhai) [reply main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: additionalProperties object is filled only by values from additionalMap object, which type is Map<String, String>

[the remark this answers is not in the document]
Comment (bchen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tSqoopImport/tSqoopImport_javaapi.javajet]: additionalMap is a Map in javajet scope, but additionalProperties_<%=cid%> is in runtime, in javajet scope the type always string as it just a textfield to store the user setting, but additionalProperties_<%=cid%> should not only accept String as value

**Pull request merged**